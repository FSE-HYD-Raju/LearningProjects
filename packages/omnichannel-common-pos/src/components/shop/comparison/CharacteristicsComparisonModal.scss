$dark-blue: #1f2f5d;
$border-color: rgb(215, 215, 215);
$blue: #4287B7;

.CharacteristicsComparisonModal {
	&-outer-container {
		margin: 0 auto 10px;
		padding: 0 10px;
		overflow: auto;
	}

	&-container {
		display: flex;
		margin: 10px auto;
		justify-content: center;
		flex-direction: column;
	}

	&-inner-container {
		display: flex;
	}

	&-item-container {
		padding: 20px;
		border: 1px solid lightgray;
	}

	&-item-container:not(:first-child) {
		border-left: none;
	}

	&-tablebody {
		padding: 0;
	}

	&-item-detail {
		border: 1px solid lightgray;
		display: inline-block;
		padding: 20px;
	}

	&-corner,
	&-corner-with-border {
		display: inline-block;
		height: 43px;
		min-width: 100px;
	}

	&-corner-with-border {
		border-top: 1px solid lightgray;
	}

	&-tableheader {
		border-left: 1px solid lightgray;
		display: inline-block;
		margin: 0;
		min-width: 160px;
		max-width: 160px;
		padding: 0 0 0 5px;
	}

	&-mugshot {
		font-weight: bold;
		height: auto;
	}

	&-character-name {
		border-top: 1px solid lightgray;
		font-weight: bold;
		height: 43px;
		flex-wrap: wrap;
		margin: 0;
		min-width: 100px;
		padding: 20px 0 20px 0;
		text-transform: capitalize;
		justify-content: space-between;
		flex: 0.5;
		display: flex;
	}

	&-item-price {
		min-height: 50px;
	}

	&-item-details {
		position: relative;
	}

	&-item-container:not(:first-child) {
		border-left: none;
	}

	&-item-name {
		text-align: center;
		font-size: 16px;
		margin: 10px auto;
		padding-bottom: 10px;
		color: darkgray;
		border-bottom: 1px solid #eee;
	}

	&-detail {
		display: flex;
		flex: 1;
		justify-content: space-around;
		border-top: 1px solid lightgray;
		border-left: 1px solid lightgray;
		font-size: 16px;
		height: 60px;
		margin: 0;
		flex-wrap: wrap;
		min-width: 160px;
		padding: 18px 0 0 5px;
	}

	&-item-to-basket {
		display: flex;
		justify-content: space-around;
		flex: 0.5;
		flex-wrap: wrap;
		border-top: 1px solid lightgray;
		border-left: 1px solid lightgray;
		margin: 0;
		min-width: 160px;
		max-width: 160px;
		padding: 5px 0 10px 0;
		text-align: center;
	}

	&-no-characteristics {
		display: flex;
		margin-top: 20px;
		padding: 10px;
		color: $blue;
		align-items: center;
		align-self: center;
		justify-content: center
	}

	&-items-container {
		display: flex;
		flex: 1;
		flex-wrap: unset;
	}

	&-instance-characteristics {
		display: flex;
		flex: 1;
	}
}
