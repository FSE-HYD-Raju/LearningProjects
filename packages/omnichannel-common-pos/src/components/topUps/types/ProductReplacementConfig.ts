interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
