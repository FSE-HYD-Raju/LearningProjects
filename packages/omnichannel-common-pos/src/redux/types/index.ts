import { ValidatePortInType } from "../../components/product/msisdn/MsisdnPortInSelector";
import { ConnectFluxReduxChannelType } from "./ConnectFluxReduxChannelType";
import { Disposers } from "./Disposers";
import { FluxAlt } from "./FluxAlt";
import { FluxReduxMap } from "./FluxReduxMap";
import { FluxStore } from "./FluxStore";
import { ReduxFluxActionMigration } from "./ReduxFluxActionMigration";
import { FluxWithActions } from "./fluxActions/FluxWithActions";

export {
	ConnectFluxReduxChannelType,
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration,
	FluxWithActions,
	ValidatePortInType
};

export * from "./utils/Omit";
export * from "./utils/HocResult";
export * from "./fluxActions";
export * from "../models/cms/cms.types";
export * from "../models/auth/auth.types";
export { CmsState } from "../models/cms/cms.reducers";
export { CmsAdminState } from "../models/cmsAdmin/cmsAdmin.reducers";
export * from "../models/auth/auth.types";
export * from "../models/service/service.types";
export * from "../models/support/support.types";
export * from "../models/comparison/comparison.types";
export * from "../models/consul/consul.types";
export * from "../models/currency/currency.types";
export * from "../models/eCare/changeSim/changeSim.types";
export * from "../models/delivery/delivery.types";
export * from "../models/digitalLife/digitalLife.types";
export * from "../models/eCare/activateSim/activateSim.types";
export * from "../models/eCare/suspension/suspension.types";
export * from "../models/feature/feature.types";
export * from "../models/lifecycle/lifecycle.types";
export * from "../models/organization/organization.types";
export * from "../models/productOfferingConfiguration/productOfferingConfiguration.types";
export * from "../models/schema/schema.types";
export * from "../models/location/location.types";
export * from "../models/eligibility/eligibility.types";
export * from "../models/productOfferings/productOfferings.types";
export * from "../models/user/user.types";
export * from "../models/basket/basket.types";
export * from "../models/customer/customer.types";
export * from "../models/sales/sales.types";
export * from "./SupportCases";
export * from "../models/versionInformation/versionInformation.types";
export * from "../models/msisdnSelection/msisdnSelection.types";
export * from "../models/portIn/portIn.types";
export * from "../models/marketingConsent/marketingConsent.types";
export * from "../models/category/category.types";
export * from "../models/b2cCheckout/b2cCheckout.types";
export { ProvinceAndCityState, Province, CityType } from "../models/provinceAndCity/provinceAndCity.types";

export * from "./routes/HasPath";
export * from "./routes/ApplicationRoute";
export * from "./HasSchema";
export * from "./HasFlux";
export * from "./HasId";
export * from "./HasPrices";
export * from "./BillingInterval";
export * from "./Cardinality";
export * from "./Characteristic";
export * from "./CharacteristicValue";
export * from "./CommercialEnrichments";
export * from "./Country";
export * from "./Links";
export * from "./FilterLabel";
export * from "./Gender";
export * from "./ImageType";
export * from "./Msisdn";
export * from "./MsisdnConfig";
export * from "./Price";
export * from "./SimplePrice";
export * from "./PriceRange";
export * from "./PriceType";
export * from "./PriceRangeBoundaries";
export * from "./ProductFilter";
export * from "./ProductOffering";
export * from "./ProductOfferingGroup";
export * from "./AlternateProductOffering";
export * from "./StockLevel";
export * from "./ProductTypes";
export * from "./ServiceUnit";
export * from "./Resource";
export * from "./SimCard";
export * from "./UnitOfMeasure";
export * from "./Allowance";
export * from "./ValidityPeriod";
export * from "./ValueRegulator";
export * from "./AutoCharacteristicConfigurationType";
export * from "./ChargingBalances";
export * from "./ChargingBalanceChanges";
export * from "./ProductsModifyConfig";
export * from "./Product";
export * from "./Agreement";
export * from "./ContactMediaType";
export * from "./EmailAddress";
export * from "./PhoneNumber";
export * from "./PostalAddress";
export * from "./Location";
export * from "./StreetAddressTokens";
export * from "./IssuingAuthority";
export * from "./Identification";
export * from "./CommonPerson";
export * from "./RelatedPerson";
export * from "./OrderLifeCycleStatus";
export * from "./Order";
export * from "./OrderCancel";
export * from "./OrderItem";
export * from "./OrderTotalsType";
export * from "./Person";
export * from "./MsisdnLifecycleStatus";
export * from "./Basket";
export * from "./payment/ContextualPayment";
export * from "./payment/ContextualPaymentMethod";
export * from "./payment/ContextualPaymentValidation";
export * from "./payment/CustomerPaymentMethod";
export * from "./payment/CustomerPaymentMethodType";
export * from "./payment/PaymentMethodsFiltersType";
export * from "./payment/CreditCard";
export * from "./payment/Role";
export * from "./BasketItem";
export * from "./SpecSubType";
export * from "./SpecType";
export * from "./TargetLifecycleStatus";
export * from "./BasketLifecycleStatus";
export * from "./BasketValidationAction";
export * from "./BasketValidationInformation";
export * from "./BasketProduct";
export * from "./PersonToPersonRelationship";
export * from "./CustomerCase";
export * from "./Category";
export * from "./SubCategory";
export * from "./ProductLifecycleStatus";
export * from "./UsageLimits";
export * from "./UsageCounters";
export * from "./Identifier";
export * from "./Notification";
export * from "./Document";
export * from "./BillingBalance";
export * from "./ShippingMethod";
export * from "./ChangeResourceInitialization";
export * from "./ChangeResource";
export * from "./Reason";
export * from "./MsisdnReservation";
export * from "./payment/PaymentStatus";
export * from "./payment/PaymentInvocation";
export * from "./Owner";
export * from "./CustomerInteraction";
export * from "./Initiator";
export * from "./InteractionDetails";
export * from "./AttachedDocument";
export * from "./Discount";
export * from "./Interval";
export * from "./BasketPaymentReceipt";
export * from "./PaginationData";
export * from "./PersonsOrderData";
export * from "./PersonsOrderState";
export * from "./OrderStatusFilter";
export * from "./SalesInfo";
export * from "./Shipment";
export * from "./AxiosResponse";
export * from "./CommercialEnrichmentName";
export * from "./UsageEvent";
export * from "./UsagePrice";
export * from "./NonMonetaryAmount";
export * from "./EventDetails";
export * from "./LifecycleChangeAction";
export * from "./Service";
export * from "./ServiceLifeCycleStatus";
export * from "./Specification";
export * from "./CustomerAccount";
export * from "./CustomerAccountLifecycleStatus";
export * from "./RecurringTopUpModelType";
export * from "../../components/priceComponent/model/PriceComponentPriceType";
export * from "./ErrorType";
export * from "./PostalAddressValidationErrorType";
export * from "./ProductModification";
export * from "./CallForwardingServiceModify";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./PortInDecision";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./DomainContext";
export * from "./ServiceModification";
export * from "./ResourceStocks";
export * from "./ResourceInventories";
export * from "./Inventory";
export * from "./SalesOrganizationRole";
export * from "./NominationCharacteristics";
export * from "./NominationPOCharacteristicsConfig";
export * from "./BasketItemsColumn";
export * from "./OrderItemsColumn";
export * from "./OrdersColumn";
export * from "./BasketsColumn";
export * from "./ConsumptionData";
export * from "./ProductsReplaceType";
export * from "./OmnichannelApiResponse";
export * from "./ProductConfiguration";
export * from "./TopUpCharacteristic";
export * from "./RecurringTopUpIntervalType";
export * from "./RecurringTopUpCharacteristicsValue";
export * from "./RecurringTopUpProductType";
export * from "./HasProduct";
export * from "./Store";
export * from "./QueryStatus";
export * from "./DeviceReservation";
export * from "./HasIntl";
export * from "./ProductUsageColumn";
export * from "./WorkforceAppointment";
