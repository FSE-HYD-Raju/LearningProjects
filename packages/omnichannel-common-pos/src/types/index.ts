import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
