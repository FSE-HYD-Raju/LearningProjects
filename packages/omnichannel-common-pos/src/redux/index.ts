"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
