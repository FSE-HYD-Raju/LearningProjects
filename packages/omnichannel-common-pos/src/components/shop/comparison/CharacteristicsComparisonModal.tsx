import cssns from "../../../utils/cssnsConfig";
import OcModal from "../../ocComponents/OcModal";
import ShopProductImage from "../product/ShopProductImage";
import ComparisonPrice from "./ComparisonPrice";
import ProductOfferingUtil from "../../../utils/ProductOfferingUtil";
import { get, isEmpty } from "lodash";
import { FormattedMessage } from "../../../channelUtils";
import messages from "./ShopComparison.messages";
import { FC } from "react";
import { BasketActionAddProductToBasket, ProductOffering, User, Configurations } from "../../../redux/types";
import { OcButton, OcButtonType } from "../../ocComponents/button/OcButton";

const { React } = cssns("CharacteristicsComparisonModal");

interface CharacteristicsComparisonModalStateProps {
	items: Array<ProductOffering>;
	open: boolean;
	comparisonCharacteristics: any;
	activeCustomer?: User;
	configurations: Configurations;
	category?: string;
	activeBasketId?: string;
}

interface CharacteristicsComparisonModalActionProps {
	actions: {
		addProductToBasket: BasketActionAddProductToBasket;
		hide: () => void;
		showConfigurationModalForProduct: (itemId: string) => void;
	};
}

type CharacteristicsComparisonModalProps = CharacteristicsComparisonModalStateProps & CharacteristicsComparisonModalActionProps;

const CharacteristicsComparisonModal: FC<CharacteristicsComparisonModalProps> = (props: CharacteristicsComparisonModalProps) => {
	const items = props.items || [];

	const show = props.open || false;
	const itemCount = items && items.length;

	const interestingCharacteristics = props.comparisonCharacteristics && props.category && props.comparisonCharacteristics[props.category];

	const handleClick = (item: any, isConfigurable: boolean) => {
		const basketId = props.activeBasketId;

		if (isConfigurable && item) {
			props.actions.showConfigurationModalForProduct(item.id);
			props.actions.hide();
		} else {
			props.actions.addProductToBasket(
				{
					product: item,
					configurations: props.configurations,
					basketId,
					hasCustomer: Boolean(props.activeCustomer)
				}
			);
		}
	};

	const renderProductNamesAndPrices = () => {
		const items = props.items || [];
		return items && items.map((item, idx) => {
			const prices = get(item.attributes || item, "prices", []);
			return (
				<div
					key={`CharacteristicsComparisonModal-tableheader-${item.id}-${idx}`}
					className="tableheader"
				>
					<div className="mugshot">
						<ShopProductImage idx={idx} product={item}/>
						{ProductOfferingUtil.getName(item)}
					</div>
					<div className="item-price">
						<ComparisonPrice
							prices={prices}
						/>
					</div>
				</div>
			);
		});
	};

	const renderConfigurables = () => {
		const items = props.items || [];
		return items && items.map((item, idx) => {
			const isConfigurable = ProductOfferingUtil.isConfigurable(item);
			const msg = isConfigurable ? messages.configure : messages.select;
			return (
				<div key={`buttons-below-item-${item.id}-${idx}`} className="item-to-basket">
					<OcButton
						buttonType={OcButtonType.SUCCESS}
						id={`pos-compare-devices-select-device-button-${item.id}`}
						onClick={() => handleClick(item, isConfigurable)}
					>
						<FormattedMessage {...msg}/>
					</OcButton>
				</div>
			);
		});
	};

	const renderCharacteristics = () => {
		const items = props.items || [];
		return (
			<div className="tablebody">
				{interestingCharacteristics && interestingCharacteristics.map((key: string, cIdx: number) => {
						const charInstances = items.map((item: any) => get(item.attributes || item, "instanceCharacteristics[" + key + "]"));
						return (
							<div className="instance-characteristics" key={`instanceCharacteristics-key-${name}-${cIdx}`}>
								<div className="character-name">
									{charInstances && charInstances.length > 0 && charInstances[0] ? (charInstances[0].name) : (key)}
								</div>
								{charInstances.map((ci: any, ciIdx: number) => {
										const value = ci && ci.values[0].value ? ci.values[0].value : " ";
										return (
											<div key={`instanceCharacteristics-value-${name}-${ciIdx}`} className="detail">
												{value}
											</div>
										);
									}
								)}
							</div>
						);
					}
				)}
			</div>
		);
	};

	return (
		<OcModal
			className="comparison"
			title={<FormattedMessage {...messages.compareProducts} />}
			showModal={show}
			largeModal={itemCount <= 3}
			extraLargeModal={itemCount > 3}
			onClose={() => { props.actions.hide(); }}
		>
			<div className="outer-container">
				<div className="container">
					<div className="inner-container">
						{renderProductNamesAndPrices()}
					</div>
					<div className="items-container">
						{renderConfigurables()}
					</div>
					{isEmpty(interestingCharacteristics) ? (
						<div className="no-characteristics">
							<FormattedMessage {...messages.detailedComparisonNotAvailable}/>
						</div>
					) : renderCharacteristics()}
				</div>
			</div>
		</OcModal>
	);
};

export default CharacteristicsComparisonModal;
export {
	CharacteristicsComparisonModalProps,
	CharacteristicsComparisonModalStateProps,
	CharacteristicsComparisonModalActionProps,
};
