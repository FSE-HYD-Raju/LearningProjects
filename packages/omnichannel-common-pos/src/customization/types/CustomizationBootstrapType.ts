import { ConnectFluxReduxChannelType, FluxAlt, StoreConfig, AppState } from "../../redux";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";
import { Store } from "react-redux";

export type GetStoreConfig<S extends AppState> = {
	reducers?: object|undefined;
	customSaga?: StoreConfig<S>["customSaga"]|null;
	interceptReducers?: StoreConfig<S>["interceptReducers"]|undefined;
	history?: StoreConfig<S>["history"]|undefined;
	initialState?: StoreConfig<S>["initialState"]|undefined;
};

export interface CustomizationBootstrapType<S extends AppState> {
	getFlux: <S>(fluxCustomization: {
		customizedStores?: Record<string, any>,
		customizedActions?: Record<string, any>,
	}) => FluxAlt<S & AppState>;
	getStore: <S>(config: GetStoreConfig<S & AppState>) => Store<S>;
	connectFluxReduxChannel: ConnectFluxReduxChannelType;
	customizationPointsMapping: CustomizationPointsMapping<S>;
}
