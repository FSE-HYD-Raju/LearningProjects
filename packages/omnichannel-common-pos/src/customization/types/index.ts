import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
