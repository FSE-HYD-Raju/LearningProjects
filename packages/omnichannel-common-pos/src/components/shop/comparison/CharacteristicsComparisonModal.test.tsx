import * as React from "react";

import {
	mountWithContext,
	shallowWithContext,
	TestUtils
} from "../../../testUtils";
import CharacteristicsComparisonModal, { CharacteristicsComparisonModalProps } from "./CharacteristicsComparisonModal";
import { ProductOffering } from "../../../redux/types";
import { Configurations } from "../../../index";

describe("CharacteristicsComparisonModal", () => {

	const minProps: CharacteristicsComparisonModalProps = {
		items: [] as any as Array<ProductOffering>,
		open: true,
		comparisonCharacteristics: {},
		configurations: {} as any as Configurations,
		category: "",
		actions: {
			addProductToBasket: jest.fn(),
			hide: jest.fn(),
			showConfigurationModalForProduct: jest.fn(),
		}
	};

	it("succeeds at shallow mount with min props", () => {
		const wrapper = shallowWithContext(<CharacteristicsComparisonModal {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount without props", () => {
		mountWithContext(<CharacteristicsComparisonModal {...minProps}/>);
	});

	it("renders given items", () => {
		const props = {
			...minProps,
			items: [
				{
					id: "oneminus-two-po",
					attributes: {
						name: "OneMinus Two",
						categories: ["device", "mobile-phone"],
						instanceCharacteristics: {},
						commercialEnrichments: [
							{
								names: {
									name: "OneMinus Two"
								}
							}
						]
					}
				},
				{
					id: "music-po",
					attributes: {
						name: "Music Streaming Service",
						categories: ["service", "entertainment", "addon"],
						instanceCharacteristics: {},
						commercialEnrichments: [
							{
									names: {
									name: "Music Streaming Service"
								}
							}
						]
					}
				},
				{
					id: "ypad-po",
					attributes: {
						name: "Appelsiini yPad Pro",
						categories: ["device", "tablet"],
						instanceCharacteristics: {},
						commercialEnrichments: [
							{
								names: {
									name: "Appelsiini yPad Pro"
								}
							}
						]
					}
				}
			] as any as Array<ProductOffering>,
			open: true
		};

		const { getModalContents } = TestUtils;

		const wrapper = mountWithContext(
			<CharacteristicsComparisonModal {...props} />
		);

		const modalContents = getModalContents(wrapper);

		const mugshots = modalContents.find(
			".CharacteristicsComparisonModal-mugshot"
		);
		const presentedMobileDevicesNames = [];
		for (let i = 0; i < mugshots.length; ++i) {
			presentedMobileDevicesNames.push(mugshots.at(i).text());
		}

		const givenMobileDevicesNames = props.items.map((item: ProductOffering) => item.attributes!.name);
		expect(presentedMobileDevicesNames.sort()).toEqual(givenMobileDevicesNames.sort());
	});
});
