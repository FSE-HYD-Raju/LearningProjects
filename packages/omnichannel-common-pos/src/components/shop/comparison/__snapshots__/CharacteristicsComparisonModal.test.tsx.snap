// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacteristicsComparisonModal succeeds at shallow mount with min props 1`] = `
<OcModal
  className="CharacteristicsComparisonModal-comparison"
  extraLargeModal={false}
  hideCloseButton={false}
  largeModal={true}
  onClose={[Function]}
  showModal={true}
  title={
    <FormattedMessageWrapper
      defaultMessage="Compare products"
      description="Shop product comparison modal compare products message"
      id="shop-comparison-modal-compare-products-message"
      wrapperType="wrapper"
    />
  }
>
  <div
    className="CharacteristicsComparisonModal-outer-container"
  >
    <div
      className="CharacteristicsComparisonModal-container"
    >
      <div
        className="CharacteristicsComparisonModal-inner-container"
      />
      <div
        className="CharacteristicsComparisonModal-items-container"
      />
      <div
        className="CharacteristicsComparisonModal-no-characteristics"
      >
        <FormattedMessageWrapper
          defaultMessage="Detailed comparison not available"
          description="No comparison characteristics available"
          id="pos-compare-devices-no-characteristics-configured"
          wrapperType="wrapper"
        />
      </div>
    </div>
  </div>
</OcModal>
`;
